src/platform_impl/linux/mod.rs

```diff
@@ -450,8 +450,18 @@ impl Window {
     }
 
     #[inline]
-    pub fn primary_monitor(&self) -> MonitorHandle {
-        x11_or_wayland!(match self; Window(window) => window.primary_monitor(); as MonitorHandle)
+    pub fn primary_monitor(&self) -> Option<RootMonitorHandle> {
+        match self {
+            #[cfg(feature = "x11")]
+            &Window::X(ref window) => {
+                let primary_monitor = window.primary_monitor();
+                Some(RootMonitorHandle {
+                    inner: MonitorHandle::X(primary_monitor),
+                })
```

Comment:
```suggestion
                let primary_monitor = MonitorHandle::X(window.primary_monitor());
                Some(RootMonitorHandle { inner: primary_monitor })
```

Just a potential style improvement suggestion.